src/components/Firebase/firebase.js

```diff
@@ -2,129 +2,260 @@ import firebase from 'firebase';
+      name,
+      date: this.getCurrentFormattedDate(),
+      round1: round,
+      round2: round,
+      round3: round,
+    })
+      .then((ref) => {
+        cb(ref.id);
+      });
   };
 
   /**
-   * @param {number} teamNumber
-   * @param {array} answers
+   * Delete a quiz by its id
+   * @param {String} quizId
    * @param {function} callback
    */
-  saveAnswersForTeam = (teamNumber, answers, callback) => {
-    let teamRef = firebase.database().ref()
-      .child(this.getCurrentFormattedDate())
-      .child('teams')
-      .child(teamNumber);
-
-    answers.forEach((answer, index) => {
-      teamRef.child(index).set(answer);
-    });
+  deleteQuiz = (id, cb) => {
+    this.db.collection(`quizzes`).doc(id).delete().then(() => cb());
+  };
 
-    callback();
+
+  /**
+   * save a whole quiz
+   * it's essential that the quiz is the whole quiz object
+   * @param {String} quizId
+   * @param {Object} quiz
+   * @param {function} callback
+   */
+  saveQuiz = (id, quiz, cb) => {
+    const docRef = this.db.collection('quizzes').doc(id);
+    docRef.set(quiz).then(() => cb());
   };
 
   /**
-   * @param {Number} teamNumber
+   * returns all quizzes in the database
    * @param {function} callback
    */
-  getAnswersForTeam = (teamNumber, callback) => {
-    let teamRef = firebase.database().ref()
-      .child(this.getCurrentFormattedDate())
-      .child('teams')
-      .child(teamNumber);
-
-    teamRef.once('value')
-      .then(snapshot => {
-        if (!snapshot.exists()) {
-          return;
+  getQuizzes = (cb) => {
+    const quizzes = [];
+    this.db.collection('quizzes').get()
+      .then((snap) => {
+        snap.forEach((doc) => {
+          quizzes.push({
+            id: doc.id,
+            data: doc.data(),
+          });
+        });
+        cb(quizzes);
+      })
+      .catch((err) => {
+        console.error('Error getting documents', err);
+      });
+  };
+
+  /**
+   * Get a quiz by its ID
+   * @param {String} quizID
+   * @param {function} callback
+   */
+  getQuiz = (id, cb) => {
+    this.db.collection('quizzes').doc(id).get()
+      .then((doc) => {
+        // if no doc, return no success so we can redirect
+        // else return the document data
+        if (!doc) {
+          cb({ success: false });
+        } else {
+          cb({ success: true, data: doc.data() });
         }
+      })
+      .catch((err) => {
+        console.error('Error getting document', err);
+        cb({ success: false });
+      });
+  };
 
-        var answers = new Array(snapshot.numChildren());
+  /**
+   * create a new game by hosting a quiz
+   * @param {String} quizId
+   */
+  hostQuiz = (id) => {
+    // store the questions here so we can easily get them
+    // and they don't change
+    this.getQuiz(id, (res) => {
+      if (!res.data) return;
 
-        snapshot.forEach(answer => {
-          answers[parseInt(answer.key)] = answer.val();
-        });
+      firebase.database().ref().set({
```

Comment:
This is using the traditional real-time database while other calls use firestore... we should probably just stick with the real-time database (like it's used here)